src/test/scala/is/hail/annotations/UnsafeSuite.scala

```diff
@@ -278,29 +279,31 @@ class UnsafeSuite extends SparkSuite {
     val rvb = new RegionValueBuilder(region)
     val rvb2 = new RegionValueBuilder(region2)
 
-    val g = Type.genStruct
+    val g = PType.genStruct
       .flatMap(t => Gen.zip(Gen.const(t), Gen.zip(t.genValue, t.genValue), arbitrary[Boolean]))
       .filter { case (t, (a1, a2), b) => a1 != null && a2 != null }
       .resize(10)
     val p = Prop.forAll(g) { case (t, (a1, a2), b) =>
 
+      val tv = t.virtualType
```

Comment:
(P)Types should be immutable.  We might consider making physical/virtualType lazy val to avoid re-consing.

Reply:
immutable meaning what?

Reply:
They don't change.  Are immutable.  Just an observation that makes lazy val implementation possible.

Reply:
er, yeah. this line was to avoid reconsing

Reply:
You don't need to change it in this PR.  Just something to consider.

Reply:
coming soon in PTypes 10!

Reply:
Sweet!  Will approve once you fix the indentation.